### PR TITLE
Update cf os

### LINKED ISSUE #2521: Spike - Update to cflinuxfs4
We will need to investigate updating our buildpack as it is unclear if this is relevant to us.

Per an email from Cloud.gov:

> Subject: Warning! Support for cflinuxfs3 is ending! Upgrade Your App to cflinuxfs4 Important Update
>
> Hello! This notification is to make you aware that you are still running `cflinuxfs3` applications in your cloud.gov organization. Your applications will soon stop working unless you take action to migrate your applications to the cflinuxfs4, stack! Please follow the directions below to migrate your applications to the cflinuxfs4, stack. ## What should you do now for existing apps? For existing applications which were created under cflinuxfs3 you will need to update the stack declaration to cflinuxfs4, there are two common ways of doing this detailed below. The options below only have to be run once for each application on cflinuxfs3 , once the stack is set for an application, it is persistent until changed with any of these two steps. - Push the app manually and specify the stack with the cf cli: cf push MY-APP -s cflinuxfs4 - Use the stack-auditor cf cli plugin to change the stack without having to push the application. Documentation for using this plugin is at https://docs.cloudfoundry.org/adminguide/stack-auditor.html#change-stacks, the basic workflow is: - Install the plugin - Use the cf cli to target the org and space for your existing application - Run cf change-stack APP-NAME cflinuxfs4 to change the app to the cflinuxfs4 stack - Each application will take about a minute or so to run the cf change-stack... command depending on the size of the droplet. ## Additional information Please refer to the deprecation notice at https://cloud.gov/2023/04/27/cflinuxfs3-deprecation-update/ which will give you additional information.
>  
> We will also be hosting live demos on how to perform this process. These live demos will be hosted on a Zoom meeting once daily from Monday (5/15/2023) until Friday (5/19/2023) with the demos occurring from 1:30 PM ET to 2 PM ET. To attend this meeting please refer to the Zoom meeting information below.
>  
> Join ZoomGov Meeting
> https://gsa.zoomgov.com/j/1602200070?pwd=SVFYU1Y5SEIzRGZ1OTZ3NlV6TitnZz09
> 
> Meeting ID: 160 220 0070
> Passcode: 954679
> One tap mobile
> +16692545252,,1602200070#,,,,*954679# US (San Jose)
> +16468287666,,1602200070#,,,,*954679# US (New York)
> 
> Dial by your location
>         +1 669 254 5252 US (San Jose)
>         +1 646 828 7666 US (New York)
>         +1 646 964 1167 US (US Spanish Line)
>         +1 415 449 4000 US (US Spanish Line)
>         +1 551 285 1373 US
>         +1 669 216 1590 US (San Jose)
> Meeting ID: 160 220 0070
> Passcode: 954679
> Find your local number: https://gsa.zoomgov.com/u/adDn6gWQaX
>  
> If you have any further issues/questions, please let us know.
>  
> Salutations,
>  
> Arsalan Haider
> Cloud.Gov Support

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Updated nginx build pack 
- Added script to automate app OS/FS upgrading for apps that dont have a regular deploy cycle
- Added workflows to circi for the upgrade
Pull request closes #2521 

## How to Test
The only way to test this is to try upgrading one of the existing frontend/backend apps in the `tanf-dev` space. You can use the `tdrs-deploy` command from `commands.sh` with appropriate arguments e.g. `tdrs-deploy sandbox`. Note sandbox has already been upgraded, so to see an actual upgrade you will want to run `tdrs-deploy` against a different environment. If the deploy succeeds, run `cf audit-stack` and verify that the environment you deployed specifies the `cflinuxfs4` OS/FS and that you can authenticate to the deployment.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] Frontend apps successfully migrate
+ [ ] Backend apps successfully migrate
+ [ ] Other apps successfully migrate

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by elipe17** [comment]
Notes: 
- The latest nginx buildpack would not work. CF would not successfully pull and unpack it hence the lower version that did work. 
- I did not run the ci workflows since the dev space has already been upgraded and I did not want to encroach on staging.

File: docs/Technical-Documentation/buildpack-changelog.md
Comment (raftmsohani) [inline docs/Technical-Documentation/buildpack-changelog.md:2]: Nit: I always like to go with one version lower than current, but if you want there is a newer buildpack version available as of May 15th ->v1.2.3

**Review by raftmsohani — APPROVED** [review]
LGTM!

File: .circleci/deployment/workflows.yml
Comment (andrew-jameson) [inline .circleci/deployment/workflows.yml]: I can't find it -- is there an evaluation where we can set these to True instead of the default False? The testing steps mention running a CLI alias `tdrs-deploy` but I don't see commands.sh updated here? Are we missing some commits?

**Review by andrew-jameson — CHANGES_REQUESTED** [review]
Either commands.sh needs to be included and/or testing steps for the deploy label need to be added, preferably the latter.

File: .circleci/deployment/workflows.yml
Comment (elipe17) [reply .circleci/deployment/workflows.yml]: Right, I wanted to get some input on this and I forgot to add it as a comment. This only ever needs to be run when we are notified of an os upgrade. I wasnt sure how we might go about that in an automated way. I can also add a command to commands.sh to effectively run this (similar to deploy), but wanted thoughts too. Maybe we can discuss in office hours?

Comment (elipe17) [reply .circleci/deployment/workflows.yml]: There is still some manual intervention required since it isnt really feasible to monitor when CF releases a new OS. But the script updates all the apps that dont get updated by the deploy scripts.

**Comment by andrew-jameson** [comment]
Note to self: ensure staging clamav is updated to cflinuxfs4 after merge to develop

**Review by andrew-jameson — APPROVED** [review]

File: .circleci/README.md
Comment (ADPennington) [inline .circleci/README.md]:
```suggestion
 - To update the apps you can either deploy each of the environments (sandbox, raft, qasp, etc) from CircleCi or you can use the `tdrs-deploy <ENVIRONMENT>` command from `commands.sh`. Assuming the buildpacks are up to date, that is all you need to do.
```

**Review by ADPennington — CHANGES_REQUESTED** [review]
lgtm pending minor spelling change

**Review by ADPennington — APPROVED** [review]
👍🏾

**Pull request merged**
